Examples/advanced_examples/real_robot_integration_advanced_demo.py

```
@@ -1,30 +1,367 @@
+        # Confirm the planned path respects joint limits before we command it.
+        pos = traj["positions"]
+        lo, hi = self.joint_limits[:, 0], self.joint_limits[:, 1]
+        within = np.all((pos >= lo - 1e-6) & (pos <= hi + 1e-6))
+        print(f"      Planned {pos.shape[0]} waypoints; "
+              f"within joint limits: {within}")
+        return traj
+
+    # ------------------------------------------------------------------ #
+    # Stage 3 + 4: closed-loop control with a forward-dynamics plant
+    # ------------------------------------------------------------------ #
+    def run_control_loop(
```

Pylint (4.0.5)

[refactor] Too many local variables (26/15)

(R0914)